src/vs/workbench/services/notification/common/notificationService.ts

```diff
@@ -24,12 +26,16 @@ export class NotificationService extends Disposable implements INotificationServ
 	private readonly _onDidRemoveNotification = this._register(new Emitter<INotification>());
 	readonly onDidRemoveNotification = this._onDidRemoveNotification.event;
 
+	private readonly _onDidSetDoNotDisturbMode = this._register(new Emitter<NotificationsDoNotDisturbMode>());
+	readonly onDidSetDoNotDisturbMode = this._onDidSetDoNotDisturbMode.event;
+
 	constructor(
 		@IStorageService private readonly storageService: IStorageService
 	) {
 		super();
 
 		this.registerListeners();
+		this.restoreDoNotDisturbModeState();
```

Comment:
I think this is not needed and would send out an event right on startup: rather anyone that depends on DND mode should ask the notification service for the mode when needed and then listen to changes going forward.

Reply:
The main reasoning for doing this was to ensure that the filters were actually set on startup/reload. Otherwise the user ends up in a state where DND is enabled but notifications will still come through. 

E.g.
https://user-images.githubusercontent.com/25163139/175860094-dd49f91f-30bd-4750-8822-4c71018af89c.mp4

Reply:
Please see https://github.com/microsoft/vscode/commit/359ee4c1b5983e8e0eef93aa53dd87a50e9b1884https://github.com/microsoft/vscode/commit/359ee4c1b5983e8e0eef93aa53dd87a50e9b1884